### PR TITLE
[Do not merge] Add code for Microsoft.Azure.Services.AppAuthentication library

### PR DESCRIPTION
**Pull request by varunsh-coder**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

Based on discussion with @shahabhijeet, creating a PR to test CI. Will wait for @shahabhijeet to review this in first week of October.

### CODE REVIEW

File: src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.IntegrationTests/Microsoft.Azure.Services.AppAuthentication.IntegrationTests.csproj [hunk 1]
```diff
@@ -0,0 +1,69 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+
```
Comment (AArnott) [inline src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.IntegrationTests/Microsoft.Azure.Services.AppAuthentication.IntegrationTests.csproj]: nit: why so many empty lines?

File: src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.IntegrationTests/Microsoft.Azure.Services.AppAuthentication.IntegrationTests.csproj [hunk 2]
```diff
@@ -0,0 +1,69 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  
+	<PropertyGroup>
+    
+		<TargetFrameworks>netcoreapp1.1;net452</TargetFrameworks>
+  
+		<SignAssembly>True</SignAssembly>
```
Comment (shahabhijeet) [inline src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.IntegrationTests/Microsoft.Azure.Services.AppAuthentication.IntegrationTests.csproj]: @varunsh-msft you  do not need this property, it's already been added as part of common properties references for the entire repo

File: src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.IntegrationTests/Microsoft.Azure.Services.AppAuthentication.IntegrationTests.csproj [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    	
+		<AssemblyOriginatorKeyFile>$(LibraryToolsFolder)\MSSharedLibKey.snk</AssemblyOriginatorKeyFile>
+    
+		<DelaySign>True</DelaySign>
```
Comment (shahabhijeet) [inline src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.IntegrationTests/Microsoft.Azure.Services.AppAuthentication.IntegrationTests.csproj]: redundant property, please remove

File: src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.IntegrationTests/Microsoft.Azure.Services.AppAuthentication.IntegrationTests.csproj [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    
+		<ProjectReference Include="..\Azure.Services.AppAuthentication.TestCommon\Microsoft.Azure.Services.AppAuthentication.TestCommon.csproj" />
+    
+		<ProjectReference Include="..\Azure.Services.AppAuthentication\Microsoft.Azure.Services.AppAuthentication.csproj" />
+  
+	</ItemGroup>
+
+  
+
+	<PropertyGroup Condition=" '$(TargetFramework)'=='net452' ">
```
Comment (shahabhijeet) [inline src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.IntegrationTests/Microsoft.Azure.Services.AppAuthentication.IntegrationTests.csproj]: the compiler constant as well as property group is part of common targets/props. Please remove

File: src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.TestCommon/Microsoft.Azure.Services.AppAuthentication.TestCommon.csproj [hunk 5]
```diff
@@ -0,0 +1,61 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  
+	<PropertyGroup>
+    
+		<TargetFrameworks>netcoreapp1.1;net452</TargetFrameworks>
+  
+		<SignAssembly>True</SignAssembly>
```
Comment (shahabhijeet) [inline src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.TestCommon/Microsoft.Azure.Services.AppAuthentication.TestCommon.csproj]: @varunsh-msft if you model your project file after one of the existing project file, that will help eliminating redundant properties.

File: src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.Unit.Tests/Microsoft.Azure.Services.AppAuthentication.Unit.Tests.csproj [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+
+
+	<ItemGroup>
+    
+		<PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0" />
+    
+		<PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
+    
+		<PackageReference Include="xunit" Version="2.2.0" />
```
Comment (shahabhijeet) [inline src/SdkCommon/AppAuthentication/Azure.Services.AppAuthentication.Unit.Tests/Microsoft.Azure.Services.AppAuthentication.Unit.Tests.csproj]: @varunsh-msft please model your test project from one of the eixsitng test project and remove redundant properties

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Comment by shahabhijeet** [comment]
@varunsh-msft as we talked earlier this week. We will revisit merging this code next month. Until then please address conflicts in this PR and reopen the PR.
Closing it for now.

**Pull request closed without merging**